backend/src/test/java/ai/verta/modeldb/TestsInit.java

```diff
@@ -165,15 +161,25 @@ public static void setServerAndService() throws Exception {
   }
 
   @AfterClass
-  public static void removeServerAndService() {
+  public static void removeServerAndService() throws InterruptedException {
     App.initiateShutdown(0);
 
-    // Remove all entities
-    // removeEntities();
-    // Delete entities by cron job
-    deleteEntitiesCron.run();
+    cleanUpResources();
 
     // shutdown test server
     serverBuilder.build().shutdownNow();
   }
+
+  protected static void cleanUpResources() throws InterruptedException {
+    // Remove all entities
+    // removeEntities();
+    // Delete entities by cron job
+    ReconcilerInitializer.softDeleteProjects.resync();
+    Thread.sleep(5000);
+    ReconcilerInitializer.softDeleteExperiments.resync();
+    Thread.sleep(5000);
```

Comment:
Time-based testing is a bad, bad idea. It both makes tests slower _and_ create flaky tests. Can we do better?